astropy/io/ascii/html.py

```diff
@@ -78,7 +81,14 @@ def process_lines(self, lines):
         """
 
         try:
-            from bs4 import BeautifulSoup
+            from bs4 import BeautifulSoup as _BeautifulSoup
+            # With Python 3.4, BeautifulSoup emits deprecation warnings over
+            # which we have no control so we wrap it.
+            class BeautifulSoup(_BeautifulSoup):
+                def __init__(self, *args, **kwargs):
+                    with warnings.catch_warnings():
+                        warnings.simplefilter("ignore", DeprecationWarning)
```

Comment:
Not sure it's worth changing this for now, but FWIW we have our own version of [`catch_warnings`](https://github.com/astropy/astropy/blob/1f0047868ca85698ca0de4c91d2f393844a830e7/astropy/tests/helper.py#L460) that's a bit nicer than the one in the `warnings` module (maybe we should move this to `astropy.utils` since I think it's useful in more contexts than just testing...)